src/basic/time-util.c

```diff
@@ -1568,3 +1572,14 @@ int time_change_fd(void) {
 
         return -errno;
 }
+
+static const char* const timestamp_style_table[_TIMESTAMP_STYLE_MAX] = {
+        [TIMESTAMP_PRETTY] = "pretty",
+        [TIMESTAMP_US] = "us",
+        [TIMESTAMP_US_UNICODE] = "µs",
+        [TIMESTAMP_UTC] = "utc",
+        [TIMESTAMP_US_UTC] = "us+utc",
+        [TIMESTAMP_US_UTC_UNICODE] = "µs+utc",
```

Comment:
uh oh, i don#t think we should have separate enums for the unicode version. I mean, there's no difference between the unicode version and the regular one, hence it should have the same low-level id, i.e. same enum value.

i.e. drop the unicode enum values, and then define the `_to_string()` and `_from_string()` functions like this:

```c
/* Only the mapping function enum → string we use the macro for */
DEFINE_STRING_TABLE_LOOKUP_TO_STRING(timestamp_style, TimestampStyle);

/* For the string → enum mapping we use the generic implementation, but then support two alias names */
TimestampStyle timestamp_style_from_string(const char *s) {
        TimestampStyle t;
        t = (TimestampStyle) string_table_lookup(timestamp_style_table, ELEMENTSOF(timestamp_style_table), s); 
        if (t >= 0)
                return t;
        if (streq_ptr(s, "µs"))
                return TIMESTAMP_US;
        if (streq_ptr(s, "µs+uts"))
                return TIMESTAMP_US_UTC;
        return t;
}
```

Reply:
thanks for the suggestion, applied